active/ItemsSourceView-Indexof/ItemsSourceView-IndexOf.md

```diff
@@ -0,0 +1,58 @@
+# Background
+<!-- Use this section to provide background context for the new API(s) 
+in this spec. -->
+The Xaml [ItemsRepeater](https://docs.microsoft.com/uwp/api/Microsoft.UI.Xaml.Controls.ItemsRepeater) control displays a list of objects using a specified template (similar to a [ListView](https://docs.microsoft.com/uwp/api/Windows.UI.Xaml.Controls.ListView)). You provide the list of objects with the [ItemsRepeater.ItemsSource](https://docs.microsoft.com/uwp/api/Microsoft.UI.Xaml.Controls.ItemsRepeater.ItemsSource) property. The list need only be an IIterable but can be an IVector for better perf. ItemsRepeater exposes a read-only [ItemsRepeater.ItemsSourceView](https://docs.microsoft.com/uwp/api/Microsoft.UI.Xaml.Controls.ItemsRepeater.ItemsSourceView) property that provides a view on this list (an [ItemsSourceView](https://docs.microsoft.com/uwp/api/Microsoft.UI.Xaml.Controls.ItemsSourceView)), which has helpers like Count and GetAt over what might be just an IIterable.
+
+Currently the ItemsSourceView class does not expose an IndexOf function. However during development, this method was used multiple times internally on derivatives of the class. To make the usage of this existing function easier on subclasses, this spec adds an IndexOf function to the ItemsSourceView.
+
+# Description
+
+The IndexOf function returns the index of the given element in the items source the ItemsSourceView is used for.
+Note that if an element is not contained in the items source, the function will return -1.
+> Spec note: returning -1 matches the behavior of for example IndexOf on [UIElementCollection](https://docs.microsoft.com/uwp/api/Windows.UI.Xaml.Controls.UIElementCollection) and List<T>.
+# Examples
+C# example
+
+```c#
+/// ************** Behavior ************ ///
+
+var items = new List<string>() { "One", "Two" };
+var itemsSourceView = new ItemsSourceView(items);
+var secondItem = itemsSourceView.GetAt(1);
+Assert.AreEqual(1, itemsSourceView.IndexOf(secondItem);
+
+
+/// ****** Use case in event handler ****** ///
+
+public void ItemInvoked(object sender, RoutedEventArgs e)
+{
+  int index = myItemsSourceView.IndexOf(sender); 
+
+  // Do something with index, here e.g. :
+  Debug.WriteLine( $"Item {index} of items source was invoked");
+}
+
+```
+
+
+# Remarks
+
+If an element is not present in the collection, the method returns -1.
+
+# API Notes
+ItemsSourceView.IndexOf(IInspectable item): Returns index in the items source. If an item is not in the items source, it will return -1.
```

Comment:
What if the item is present more than once?

Option 1: You get the lowest-index one.
Option 2: You get the highest-index one.
Option 3: You get an arbitrary one.

Reply:
Looking at the implementation, this is nothing we have control over as we rely on the underlying collection to do this. So if the underlying collection returns the lowest index, so do we.

Reply:
Should we mention this in the documentation (and should add to the spec)?

Reply:
In the case of an IIterable, we create a vector from the IIterable by iterating over the collection and adding it to an empty winrt::vector, so we would use winrt::vector::IndexOf .

Reply:
This was one of the reasons we did not add it initially, but later found out that returning the lowest-index one is reasonable, and in cases where there are no duplicates, this is a non issue. @chingucoding is correct about collections that are just Iterable, we iterate over it and make a copy. Where the provided collection supports IndexOf, it will be forwarded to its implementation.

Reply:
Added new suggested text in a separate comment to cover this.